array.c

```diff
@@ -1274,20 +1321,63 @@ rb_ary_pop(VALUE ary)
 
 /*
  *  call-seq:
- *     ary.pop    -> obj or nil
- *     ary.pop(n) -> new_ary
+ *    ary.pop -> obj or nil
+ *    ary.pop(n) -> new_array
  *
- *  Removes the last element from +self+ and returns it, or
- *  +nil+ if the array is empty.
+ *  Removes and returns trailing elements from <tt>ary</tt>.
  *
- *  If a number +n+ is given, returns an array of the last +n+ elements
- *  (or less) just like <code>array.slice!(-n, n)</code> does. See also
- *  Array#push for the opposite effect.
+ *  Argument <tt>n</tt>, if given, must be an
+ *  {Integer-convertible object}[doc/implicit_conversion_rdoc.html#label-Integer-Convertible+Objects].
  *
- *     a = [ "a", "b", "c", "d" ]
- *     a.pop     #=> "d"
- *     a.pop(2)  #=> ["b", "c"]
- *     a         #=> ["a"]
+ *  ---
+ *
+ *  When no argument is given,
+ *  removes and returns the last element in <tt>ary</tt>
+ *  unless <tt>ary</tt> is empty:
+ *
+ *    a = [:foo, 'bar', 2]
+ *    a.pop # => 2
+ *    a # => [:foo, "bar"]
+ *
+ *  Returns <tt>nil</tt> if <tt>ary</tt> is empty:
+ *
+ *    a = []
+ *    a.pop # => nil
+ *
+ *  ---
+ *
+ *  When argument <tt>n</tt> is given and <tt>0 <= n <= ary.size</tt>,
+ *  removes and returns the last <tt>n</tt> elements in a new \Array:
+ *
+ *    a = [:foo, 'bar', 2]
+ *    a1 = a.pop(2)
+ *    a1 # => ["bar", 2]
+ *    a # => [:foo]
+ *    a.pop(0) # => []
+ *
+ *  If <tt>n > ary.size</tt>,
+ *  removes and returns all elements in <tt>ary</tt>:
```

Comment:
I think we can drop "in +ary+", there might be some confusion with the example below that uses `a` instead of `ary`.

Reply:
Done.